ui_testing/testcases/header_tests/test08_usermanagement.py

```diff
@@ -238,4 +221,260 @@ def test010_update_user_role_with_save_cancel_btn(self, save):
+
+
+class LoginRandomUser(BaseTest):
+
+    def setUp(self):
+        super().setUp()
+        random_user_name = self.generate_random_string()
+        random_user_email = self.base_page.generate_random_email()
+        random_user_password = self.generate_random_string()
+        self.users_api.create_new_user(random_user_name, random_user_email, random_user_password)
+        self.login_page.login(username=random_user_name, password=random_user_password)
+        self.base_selenium.wait_until_page_url_has(text='dashboard')
+        self.header_page.click_on_header_button()
+
+    def test018_delete_user_used_in_other_entity(self):
+        """
+        User management: Make sure that you can't delete any user record If this record used in other entity
+        LIMS-6407
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+
+        last_row = self.header_page.get_last_user_row()
+        self.header_page.click_check_box(source=last_row)
+
+        self.header_page.archive_entity(menu_element='user_management:right_menu',
+                                        archive_element='user_management:archive')
+        self.header_page.get_archived_entities(menu_element='user_management:right_menu',
+                                               archived_element='user_management:archived')
+
+        last_row = self.header_page.get_last_user_row()
+        self.header_page.click_check_box(source=last_row)
+        self.header_page.delete_entity()
+        self.assertTrue(self.base_selenium.element_is_displayed(element='general:confirmation_pop_up'))
+
+    def test019_filter_by_changed_by(self):
+        """
+        Header: Roles & Permissions Approach: Make sure that you can filter by role changed by
+        LIMS-6507
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+
+        new_user = self.generate_random_string()
+        new_email = self.base_page.generate_random_email()
+        self.header_page.create_new_user(user_name=new_user, user_email=new_email,
+                                         user_role='Admin', user_password='1', user_confirm_password='1')
+
+        self.header_page.click_on_user_config_btn()
+        self.base_selenium.click(element='user_management:checked_changed_by')
+        self.base_selenium.click(element='user_management:apply_btn')
+
+        last_row = self.header_page.get_last_user_row()
+        row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=last_row)
+        user_changed_by = row_data['Changed By']
+
+        self.base_selenium.click(element='general:menu_filter_view')
+        self.header_page.filter_user_drop_down(filter_name='user_management:filter_changed_by',
+                                               filter_text=user_changed_by)
+
+        users_result = self.header_page.get_table_rows_data()
+        self.assertIn(row_data['Changed By'], users_result[0])
```

Comment:
```suggestion
        self.assertIn(new_user, users_result[0])
```

Reply:
@0xIslamTaha assert is not correct here I will not use new_user at all ??